### PR TITLE
Adapt API Access tests after new changes in dashboard endpoints

### PR DESCRIPTION
**Pull request by bdattoma**
Adapting tests after changes introduced by https://issues.redhat.com/browse/RHODS-5388 
Main changes:
- `pvc` endpoint is not reachable anymore: tests expects 404
- `notebooks` endpoints have changed the expected JSON body to create/update/stop notebook servers plus PUT has been removed: tests use POST for both creation and update, while PATCH is used to stop
- `configmaps` and `secrets` endpoints have been replaced by `envs` which should allow GET only

Moreover it adds the test for `route` endpoint which was introduced in v1.16

Signed-off-by: bdattoma [bdattoma@redhat.com](mailto:bdattoma@redhat.com)

### CODE REVIEW

**Review by lugi0 — DISMISSED** [review]

**Comment by bdattoma** [comment]
Jenkins job: [1331](https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/1331/console) _SUCCESS_

File: tests/Tests/400__ods_dashboard/414__ods_dashboard_api.robot [hunk 1]
```diff
@@ -776,15 +831,35 @@ Set Username In PVC Payload
...
+    [Documentation]     Fill in the json body for creating/updating a Notebook with the username
+    [Arguments]     ${notebook_username}=${NONE}
+    IF    "${notebook_username}" == "${NONE}"
+        ${complete_body}=       Set Variable      ${NB_STOP_ENDPOINT_BODY_A}
+    ELSE
+        ${complete_body}=       Replace String    ${NB_STOP_ENDPOINT_BODY_B}    <USERNAME>    ${notebook_username}
+    END
+    [Return]    ${complete_body}
 
 Delete Test PVCs
```
Comment (jgarciao) [inline tests/Tests/400__ods_dashboard/414__ods_dashboard_api.robot:855]: In tests/Resources/Page/ODH/JupyterHub/PVCChange.resource there is a keyword named `May Be Delete PVC` to delete one PVC. 

Would you mind (in this PR or another one) creating a new file OCPDashboard/Storage/Storage.robot and placing both keywords (could be renamed to `Delete PVC` and `Delete PVC List` or similar

**Review by jgarciao — APPROVED** [review]
lgtm (with suggestion for another PR)

File: tests/Tests/400__ods_dashboard/414__ods_dashboard_api.robot (hunk 1 above)
Comment (bdattoma) [reply tests/Tests/400__ods_dashboard/414__ods_dashboard_api.robot:855]: okay will do in another PR

**Pull request merged**